### PR TITLE
feat(api): Decision Notification Listener for feature variable APIs.

### PR DESCRIPTION
**Pull request by mfahadahmed**
## Summary
- Implemented Decision Notification Listener for feature variable APIs.

## Test plan
- Added unit tests.

### CODE REVIEW

File: OptimizelySDK/Optimizely.cs [hunk 1]
```diff
@@ -655,5 +646,33 @@ private bool ValidateStringInputs(Dictionary<string, string> inputs)
...
+        {
+            object result = null;
+            switch (type)
+            {
+                case FeatureVariable.VariableType.BOOLEAN:
+                    bool.TryParse(value, out bool booleanValue);
+                    result = booleanValue;
+                    break;
+                case FeatureVariable.VariableType.DOUBLE:
+                    double.TryParse(value, out double doubleValue);
```
Comment (mikeng13) [inline OptimizelySDK/Optimizely.cs]: Does it matter that in the previous code we used the `Double` class to parse and here we are calling it from the `double` primitive type? 

`Double.TryParse(variableValue, out double doubleValue)`

File: OptimizelySDK/Optimizely.cs [hunk 2]
```diff
@@ -655,5 +646,33 @@ private bool ValidateStringInputs(Dictionary<string, string> inputs)
...
+                    result = doubleValue;
+                    break;
+                case FeatureVariable.VariableType.INTEGER:
+                    int.TryParse(value, out int intValue);
```
Comment (mikeng13) [inline OptimizelySDK/Optimizely.cs]: Same question as above, where we use to parse from the `Int32` class

File: OptimizelySDK/Optimizely.cs [hunk 3]
```diff
@@ -406,7 +406,7 @@ public virtual bool IsFeatureEnabled(string featureKey, string userId, UserAttri
         /// <param name="userAttributes">The user's attributes</param>
         /// <param name="variableType">Variable type</param>
         /// <returns>string | null Feature variable value</returns>
-        public virtual string GetFeatureVariableValueForType(string featureKey, string variableKey, string userId, 
+        public virtual T GetFeatureVariableValueForType<T>(string featureKey, string variableKey, string userId,
```
Comment (mikeng13) [inline OptimizelySDK/Optimizely.cs]: This is an interesting improvement :)

File: OptimizelySDK/Optimizely.cs (hunk 1 above)
Comment (msohailhussain) [reply OptimizelySDK/Optimizely.cs]: https://stackoverflow.com/questions/3951224/is-there-a-meaningful-difference-between-double-and-double-in-net
No difference.

File: OptimizelySDK/Optimizely.cs (hunk 3 above)
Comment (msohailhussain) [reply OptimizelySDK/Optimizely.cs]: yeah, we actually needed to notify for every featurevariable* api, and type was required, thats why used generics here.

**Pull request merged**